### PR TITLE
feat: Add trust anchor interface

### PR DESCRIPTION
**Pull request by grausof**
Adds the interface to talk to the trust anchor and get the entity configuration.

#### List of Changes
- Added trust anchor infra
- Added trust anchor types

#### Motivation and Context
Useful interfaces for obtaining the trust chain

#### How Has This Been Tested?
Tested locally

#### Checklist:

- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/io-func-wallet-solution-backend/src/infra/trust-anchor/index.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      TE.map(jose.decodeJwt),
+      TE.chainEitherKW(
+        validate(
+          TrustAnchorEntityConfigurationPayload,
+          "Invalid trust anchor entity configuration"
+        )
+      ),
+      TE.map((metadata) => metadata.jwks.keys)
+    );
+  };
```
Comment (hevelius) [inline apps/io-func-wallet-solution-backend/src/infra/trust-anchor/index.ts:55]: What do you think about this refactor?

```suggestion
   getPublicKeys = () =>
    pipe(
      new URL(oidFederation, this.#configuration.trustAnchorUri.href),
      (metadataUrl) => metadataUrl.href,
      getRequest(this.fetchWithTimeout),
      TE.map(jose.decodeJwt),
      TE.chainEitherKW(
        validate(
          TrustAnchorEntityConfigurationPayload,
          "Invalid trust anchor entity configuration"
        )
      ),
      TE.map((metadata) => metadata.jwks.keys)
    );
```

File: apps/io-func-wallet-solution-backend/src/infra/trust-anchor/index.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  EntityStatementHeader,
+  EntityStatementPayload,
+  TrustAnchor,
+  TrustAnchorEntityConfigurationPayload,
+} from "../../trust-anchor";
+import { FederationEntityMetadata } from "../../entity-configuration";
+import { validate } from "../../validation";
+import { getKeyByKid } from "../../jwk";
+import { verifyJwtSignature } from "../../verifier";
+
```
Comment (hevelius) [inline apps/io-func-wallet-solution-backend/src/infra/trust-anchor/index.ts:24]:
```suggestion

const oidFederation = "/.well-known/openid-federation";

```

File: apps/io-func-wallet-solution-backend/src/infra/trust-anchor/index.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+      "anchor",
+      this.#configuration.trustAnchorUri.href
+    );
+
+    return pipe(
+      fetchUrl.href,
+      getRequest(this.fetchWithTimeout),
+      TE.chain(this.validateEntityStatementJwt)
+    );
+  };
```
Comment (hevelius) [inline apps/io-func-wallet-solution-backend/src/infra/trust-anchor/index.ts:70]: Another way to write this block. btw I guess it doesn't bring any concrete improvement.

```suggestion
 getEntityStatement = () =>
    pipe(
      this.#configuration.trustAnchorUri.href,
      (href) => {
        const fetchUrl = new URL("fetch", href);
        fetchUrl.searchParams.append("sub", this.#configuration.basePath.href);
        fetchUrl.searchParams.append(
          "anchor",
          this.#configuration.trustAnchorUri.href
        );
        return fetchUrl.href;
      },
      getRequest(this.fetchWithTimeout),
      TE.chain(this.validateEntityStatementJwt)
    );
```

**Comment by grausof** [comment]
@hevelius Thanks, I applied your suggestions.

**Review by hevelius — APPROVED** [review]
LGTM

**Pull request merged**